src/client/pythonEnvironments/legacyIOC.ts

```diff
@@ -167,6 +168,25 @@ class ComponentAdapter implements IComponentAdapter, IExtensionSingleActivationS
         );
     }
 
+    // VirtualEnvironmentPrompt
+
+    // Call callback if an environment gets created within the resource provided.
+    public onDidCreate(resource: Resource, callback: () => void): vscode.Disposable | undefined {
+        const workspaceFolder = resource ? vscode.workspace.getWorkspaceFolder(resource) : undefined;
+        return this.enabled
+            ? this.api.onChanged((e) => {
+                  if (
+                      e.type === FileChangeType.Created &&
+                      workspaceFolder &&
+                      e.searchLocation &&
+                      isParentPath(e.searchLocation.fsPath, workspaceFolder.uri.fsPath)
+                  ) {
+                      callback();
+                  }
+              })
+            : undefined;
+    }
```

Comment:
Consider breaking those code into more discrete logical chunks:

```suggestion
    public onDidCreate(resource: Resource, callback: () => void): vscode.Disposable | undefined {
        if (!this.enabled) {
            return undefined;
        }
        if (!resource) {
            return { dispose: () => undefined };
        }
        const workspaceFolder = vscode.workspace.getWorkspaceFolder(resource);
        if (!workspaceFolder) {
            return { dispose: () => undefined };            
        }
        return this.api.onChanged((e) => {
            if (e.type !== FileChangeType.Created) {
                return;
            }
            // if (e.kind && !kinds.includes(e.kind)) {
            //     return;
            // |
            if (!e.searchLocation || !isParentPath(e.searchLocation.fsPath, workspaceFolder.uri.fsPath) {
                return;
            }
            callback();
        })
    }
```

Reply:
I find the pattern you suggested a bit too discrete but you're right the quadruple condition is a bit dense to read. I refactored the code midway.